src/common.h

```diff
@@ -103,8 +99,13 @@ enum class Result {
   Error,
 };
 
-#define WABT_SUCCEEDED(x) ((x) == ::wabt::Result::Ok)
-#define WABT_FAILED(x) ((x) == ::wabt::Result::Error)
+template<typename T>
+void ZeroMemory(T& v) {
+  memset(&v, 0, sizeof(v));
+}
+
+inline bool Succeeded(Result result) { return result == Result::Ok; }
```

Comment:
I'll leave it as is for now. but will change latter to a bool. I just find the needing to test with macros, and that Result::Ok is zero, too easy to make a mistake.